### PR TITLE
Enable 2-channel arrays in resize tests

### PR DESCRIPTION
**Pull request by mshabunin**
relates #15257

### CODE REVIEW

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Comment by asmorkalov** [comment]
@mshabunin do you plan to finish the pr?

File: modules/ts/src/ts.cpp [hunk 1]
```diff
@@ -621,20 +621,27 @@ void TS::set_gtest_status()
 
 void TS::update_context( BaseTest* test, int test_case_idx, bool update_ts_context )
 {
+    CV_UNUSED(update_ts_context);
```
[the remark this answers is not in the document]
Comment (alalek) [reply modules/ts/src/ts.cpp:624]: These changes are used to have reproducible initial RNG seed (and different for each test_case_idx of the legacy test system).

They are changed by this modification, so some tests are started to fail. This is the question to the testing code.

**Comment by mshabunin** [comment]
The PR have passed all checks now. Following is the list of changes:
-  enabled 2-channel arrays in imgwarp tests (the main purpose of this PR) - this change caused some tests failure due to changed RNG state, so we...
- enabled RNG resetting for old-style (e.g. based on ArrayTest) tests to make them more stable
- increased test threshold in test_convhull.cpp (MinTriangle), test_color.cpp (Lab, Luv), test_homohraphy.cpp (`max_diff`)
- partially converted `Calib3d_Homography.accuracy` to new style (plain GTest) and `cv::theRNG()`

**Comment by alalek** [comment]
@mshabunin Please squash and rebase commits to enable GitHub Actions checks too (GHA fetch workflows from source branch only :\ ).

**Comment by mshabunin** [comment]
It should be good now - I've rebased and "fixed" one flaky test in _video_ module.

**Review by alalek — APPROVED** [review]

**Pull request merged**